File: Java/Q34.java
import java.util.Scanner;

public class SumOfTwoPrimes {

    // Function to check if a number is prime
    static boolean isPrime(int num) {
        if (num <= 1)
            return false;
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0)
                return false;
        }
        return true;
    }

    // Function to check if a number can be expressed as sum of two primes
    static void checkSumOfTwoPrimes(int number) {
        boolean found = false;
        System.out.println("Checking if " + number + " can be expressed as the sum of two prime numbers...");

        for (int i = 2; i <= number / 2; i++) {
            int complement = number - i;

            if (isPrime(i) && isPrime(complement)) {
                System.out.println(number + " = " + i + " + " + complement);
                found = true;
            }
        }

        if (!found) {
            System.out.println(number + " cannot be expressed as the sum of two prime numbers.");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Input from user
        System.out.print("Enter a number: ");
        int number = scanner.nextInt();

        // Check and display result
        checkSumOfTwoPrimes(number);

        scanner.close();
    }
}

File: Python/Q34.py
# Q34.py - Program to Check Whether a Number can be Expressed as Sum of Two Prime Numbers
def is_prime(n):
    if n <= 1: return False
    for i in range(2, int(n**0.5)+1):
        if n % i == 0:
            return False
    return True

num = int(input("Enter a number: "))
found = False
for i in range(2, num//2+1):
    if is_prime(i) and is_prime(num-i):
        print(f"{num} = {i} + {num-i}")
        found = True
        break
if not found:
    print(f"{num} cannot be expressed as sum of two primes")